src/TaskExtension.cpp

```diff
@@ -0,0 +1,107 @@
+#include "TaskExtension.h"
+
+#include <QMenu>
+#include <QRegularExpression>
+
+#include "LinkSyntaxHighlighter.h"
+
+static const char INSERT_TASK_REGEX[] = "^(\\s*)([-\\*])?(\\s)?(\\[[x ]\\])?";
+
+static int getTaskCheckmarkPosUnderCursor(const QTextCursor& cursor) {
+    return LinkSyntaxHighlighter::getTaskCheckmarkPosAt(cursor.block().text(),
+                                                        cursor.positionInBlock());
+}
+
+static void toggleTask(QTextCursor* cursor, int pos) {
+    // First insert the new character after the old one, then delete the old character.
+    // This prevents the main cursor from moving as a side effect from the editing.
+    cursor->beginEditBlock();
+    cursor->setPosition(pos + cursor->block().position());
+    cursor->movePosition(QTextCursor::Right, QTextCursor::KeepAnchor);
+    QString c = cursor->selectedText();
```

Comment:
I prefer avoiding one character variables. Can you rename this one `text`?

Reply:
Sure, I'll change it.